code/modules/power/supermatter/supermatter.dm

```diff
@@ -630,6 +642,14 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 			power_transmission_bonus += gas_comp[gasID] * gas_trans[gasID] * (isnull(transit_mod[gasID]) ? 1 : transit_mod[gasID])
 		power_transmission_bonus *= h2obonus
 
+		//Miasma is really just microscopic particulate. It gets consumed like anything else that touches the crystal.
+		if(gas_comp[/datum/gas/miasma])
+			var/miasma_pp = env.return_pressure() * gas_comp[/datum/gas/miasma]
```

Comment:
Aren't partial pressures representative of the pressure of the gas mix that's given by the molar count passed in? What the heck is this?

Reply:
Last I checked the volume of the removed gas mixture is pretty much arbitrary. As a result I can't trust that it has the right pressure values. To fix this I'm drawing on the pressure of the surf's mix which should be more accurate. Because the partial pressures of gases in a mixture are independent O can then just multiply the total pressure with the mole ratio of miasma to get the miasma's partial pressure.

Reply:
Ah I forgot what gas_comp was. You'd think I'd remember my own code eh?